### DISCUSSION TITLE
Suggestion  for search api  and select api

### DISCUSSION THREAD

**Discussion by abhijitkumar84** [ONDC-Official/mobility-specification#23] [General]
Currently below is flow for metro use case 
      1. Call search api to get all station 
      2. user will select to and from station then again call search to get fare 
      3.  After user select(SJT, RJT, PASS ...) Call select , init and confirm call 
      
Mostly metro station data is static will change very rarely , so most of bap will cache this data for better user exprience 
If Bap is caching this data  Unncessarily we will be calling  2nd search call (it defeats puprouse of caching)

proposal
  1. In search response, we would be gettign all station for particular metro and type of product that metro sell(Qr ticket, pass , smart card recharge) 
  2. can we add capbiltiy to call direct select call with to and from station , nos of ticket and in on select response we would be getting ticket fare of sjt, rjt (in case of QR), 45 trip pass, 30 trip pass (in case product type is pass)   ? we will be reducing one hop here (this is similar to fulfillment option in select retail call, in on select user call multiple option of fulfillment and user had to select it  )

**Comment by sreenivasbandari** [comment]
@92shreyansh
